autosubmit/notifications/mail_notifier.py

```diff
@@ -20,18 +20,30 @@
 import smtplib
 import email.utils
 from email.mime.text import MIMEText
+from email.mime.multipart import MIMEMultipart
+from email.mime.application import MIMEApplication
 from log.log import Log
+from pathlib import Path
+from autosubmitconfigparser.config.basicconfig import BasicConfig
 
 class MailNotifier:
     def __init__(self, basic_config):
         self.config = basic_config
 
     def notify_experiment_status(self, exp_id,mail_to,platform):
         message_text = self._generate_message_experiment_status(exp_id, platform)
-        message = MIMEText(message_text)
+        message = MIMEMultipart() 
         message['From'] = email.utils.formataddr(('Autosubmit', self.config.MAIL_FROM))
-        message['Subject'] = f'[Autosubmit] Warning a remote platform is malfunctioning'
+        message['Subject'] = f'[Autosubmit] Warning: a remote platform is malfunctioning'
         message['Date'] = email.utils.formatdate(localtime=True)
+        message.attach(MIMEText(message_text))
+
+        try:
+            files = [f for f in BasicConfig.expid_aslog_dir(exp_id).glob('*_run.log') if Path(f).is_file()]
+            self._attach_files(message, files)
+        except BaseException as e:
+            Log.printlog('An error has occurred while attaching log files to a warning email about remote_platforms ', 6011)
```

Comment:
So if we fail to attach any file, we just print the log? Or should we move this into the for-loop in attach_files? (i.e. log when a file fails, but keep trying the remaining ones?)

And should we just log, or fail the process?

And should we perhaps log but also add something in the email/mime-message, in the notification saying "Here's your notification, but sorry I couldn't attach file ABC.log for reasons"? :grimacing:

Reply:
I think ideally we'd want to attach as many log files as possible; so omit the ones we can't retrieve (if any) but still attach the rest. Modifying the mail body to explain this would also be nice, i.e. "This email has attached all the log files we could retrieve, there might be some missing". 
I can add another log in the `attach_files` function and write more specific log messages. I'd rather have a new one than moving this one since they can catch different exceptions - for instance, what if the ASLOGS dir doesn't exist? -.

I'm also wondering if we should compress the log files before attaching them, to delay reaching the file attachment limit (what is it?) Overall, attaching the files is a nice touch but the most relevant information for the user is the log path, which is always printed in the mail body.

Reply:
If any of these logs include things like compilation of a model like IFS with debug info, that might be a very-very-very large file (I've seen >100MB when someone enabled some print+debug in the compilation). Even zipping it could fail to send notifications due to server size limit.

IMHO it would be safer to add location, and if we want to give the file for the user convenience, then maybe a toggle that enables the files to be attached.